### PR TITLE
feat(admin): court pace advisor panel for schedule rebalancing (bracket-creator-8r3)

### PR DESCRIPTION
**Pull request by gitrgoliveira**
## Summary

Closes bracket-creator-8r3 — adds an admin-only **Court pace** collapsible panel to `AdminSchedulePage` that surfaces per-court progress vs planned schedule so the tournament manager can decide whether to rebalance matches between courts.

- New `computeCourtPaceStats(byCourt, perMatchMinutes, nowMinutes)` pure helper: derives `completedCount`, `remainingCount`, `wallClockElapsedMin`, `estimatedRemainingMin`, `plannedRemainingMin`, and `delta` for each court
- `CourtPacePanel` component: collapsible card with per-court status chips color-coded green (≤5 min)/amber (5–20 min)/red (>20 min) and a "Suggestion: move N matches from Shiaijo X to Y" hint when rebalancing is worthwhile
- `suggestRebalances(stats, ppm)`: pure heuristic — finds slowest+fastest courts and recommends `floor(min(slow.delta, |fast.delta|) / ppm)` moves
- Admin-only: rendered only in `AdminSchedulePage`; no changes to `viewer.jsx` or `display.jsx`
- Live updates automatically because the component rereads `byCourt` on every `match_updated` SSE event (reuses existing hook)

## Review-round fix (commit e8966a4)

- Filter out empty-court buckets at the top of `CourtPacePanel` so configured-but-unused courts (and every non-A court under `?court=A` scope) don't render "0/0 done · Done" tiles.  Both the card grid and `suggestRebalances` now operate on `populatedStats`.
- Regression test pinning that `computeCourtPaceStats` still emits rows for empty buckets — the component owns the populated filter so the helper contract stays simple.

## Test plan

- [x] 664/664 vitest tests pass (`npm test -- --run`) — 18 original + 1 new regression test
- [x] 18 new unit tests cover `computeCourtPaceStats` edge cases (empty court, no scheduledAt times, behind/ahead/on-track delta math) and `suggestRebalances` (null inputs, no imbalance, correct move count)
- [x] `make go/build` succeeds (binary built clean)
- [x] `go test ./...` — all Go packages pass
- [x] `make go/test-race` — race-detector clean (engine 88.2%, mobileapp 84.4%, etc.)
- [x] `make go/lint` — 0 issues
- [x] `npm run lint` — clean (no warnings)
- [x] Verified no leak: `grep -n CourtPacePanel web-mobile/js/viewer.jsx web-mobile/js/display.jsx` → nothing (component never exposed via `window.*`)
- [x] `make run-mobile` → open admin schedule page → confirm "Court pace" card appears when at least one court has matches
- [x] Mark several matches on Court A complete → confirm Court A shows "ahead" status (4/50 done · 580m ahead)
- [x] Verify Court pace card does NOT appear on `/viewer` or `/display/...` — confirmed absent on both routes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: web-mobile/js/admin_schedule.jsx [hunk 1]
```diff
@@ -845,6 +847,153 @@ function PerCourtBreakdown({ perCourtMinutes }) {
...
+
+// CourtPacePanel — admin-only collapsible card showing per-court pace status
+// and a rebalancing suggestion. Never rendered in viewer or display views.
+function CourtPacePanel({ byCourt, safeMatchDuration }) {
+  const [open, setOpen] = useStateA(false);
+
+  const stats = useMemoA(
+    () => computeCourtPaceStats(byCourt, safeMatchDuration),
+    [byCourt, safeMatchDuration]
+  );
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_schedule.jsx]: Fixed in 84610d5: dropped the `useMemoA` wrapper and call `computeCourtPaceStats(byCourt, safeMatchDuration)` directly. Since `byCourt` is rebuilt on every parent render, the memo dependency was always invalidated anyway — removing it makes the intent explicit.

File: web-mobile/js/admin_schedule.jsx [hunk 2]
```diff
@@ -845,6 +847,153 @@ function PerCourtBreakdown({ perCourtMinutes }) {
...
+//                           court. Falls back to estimatedRemainingMin when
+//                           no scheduled times exist.
+//   delta               — estimatedRemainingMin − plannedRemainingMin
+//                         positive = behind schedule, negative = ahead
+//
+// Exported for the vitest suite.
+export function computeCourtPaceStats(byCourt, perMatchMinutes, nowMinutes) {
+  const ppm = perMatchMinutes > 0 ? perMatchMinutes : 3;
+  const nowMin = nowMinutes !== undefined ? nowMinutes : (() => {
+    const d = new Date();
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_schedule.jsx:933]: Fixed in 84610d5: updated the comment from 'pure helper' to 'deterministic when `nowMinutes` is passed; uses wall-clock as fallback when omitted' to accurately reflect the non-deterministic default.

File: web-mobile/js/admin_schedule.jsx [hunk 3]
```diff
@@ -845,6 +847,153 @@ function PerCourtBreakdown({ perCourtMinutes }) {
...
+    return <span style={badgeStyle(stat.delta)}>{Math.round(abs)}m {dir}</span>;
+  };
+
+  return (
+    <div className="card" style={{ marginBottom: 20 }} data-testid="court-pace-panel">
+      <div
+        className="card__title"
+        style={{ display: "flex", justifyContent: "space-between", cursor: "pointer", marginBottom: open ? 12 : 0 }}
+        onClick={() => setOpen(!open)}
+        role="button"
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_schedule.jsx:1019]: Fixed in 84610d5: added `tabIndex={0}` and `onKeyDown` (Enter/Space toggle) to the collapsible header div so it is reachable by keyboard, matching the pattern used in `admin_pools.jsx` and `admin_shell.jsx`.

File: web-mobile/js/admin_schedule.jsx [hunk 4]
```diff
@@ -845,6 +847,158 @@ function PerCourtBreakdown({ perCourtMinutes }) {
   );
 }
 
+// computeCourtPaceStats(byCourt, perMatchMinutes[, nowMinutes]) — deterministic helper.
+// When nowMinutes is omitted the function falls back to wall-clock time (non-deterministic).
+// Pass an explicit nowMinutes (e.g. Date.now()/60000) for deterministic tests.
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_schedule.jsx]: Fixed in commit 3e4eb00: updated the example to new Date().getHours()*60 + new Date().getMinutes() to match the minutes-since-midnight unit the function actually expects.

File: web-mobile/js/admin_schedule.jsx [hunk 5]
```diff
@@ -845,6 +847,158 @@ function PerCourtBreakdown({ perCourtMinutes }) {
...
+// For each court, derive:
+//   court               — the court label (e.g. "A")
+//   completedCount      — matches with status === "completed"
+//   remainingCount      — matches NOT yet completed
+//   wallClockElapsedMin — minutes from the earliest scheduledAt on that court
+//                         to now. Falls back to 0 when no scheduledAt data.
+//   estimatedRemainingMin — remainingCount × perMatchMinutes
+//   plannedRemainingMin   — time from now to the latest scheduledAt on the
+//                           court. Falls back to estimatedRemainingMin when
+//                           no scheduled times exist.
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_schedule.jsx]: Fixed in 5fd2407. Docstring now mentions the `+ perMatchMinutes` (end-of-last-match) behavior so the unit contract is unambiguous.

File: web-mobile/js/admin_schedule.jsx [hunk 6]
```diff
@@ -378,6 +378,8 @@ function AdminSchedulePage({ tournament, onBack, onMoveCourt, onLogout, onViewer
           )}
         </div>
 
+        <CourtPacePanel byCourt={byCourt} safeMatchDuration={safeMatchDuration} />
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_schedule.jsx]: Fixed in b83de19. Built a separate `paceByCourt` bucket directly from `allMatches` (still narrowed by the explicit `?court=` scope so admins can review one court at a time, but NOT by the ad-hoc player/dojo/competition filters). `CourtPacePanel` now reads that instead of the filter-affected `byCourt`.

File: web-mobile/js/admin_schedule.jsx [hunk 7]
```diff
@@ -845,6 +858,159 @@ function PerCourtBreakdown({ perCourtMinutes }) {
...
+  });
+}
+
+// CourtPacePanel — admin-only collapsible card showing per-court pace status
+// and a rebalancing suggestion. Never rendered in viewer or display views.
+function CourtPacePanel({ byCourt, safeMatchDuration }) {
+  const [open, setOpen] = useStateA(false);
+
+  const stats = computeCourtPaceStats(byCourt, safeMatchDuration);
+
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_schedule.jsx:986]: Already covered by PR #149 (mp-pb1), which adds a 60s setInterval tick inside `CourtPacePanel` so the panel re-derives wall-clock-dependent stats every minute even without SSE traffic. That PR explicitly leaves `nowMinutes` undefined in the production call site (so the helper falls back to wall-clock) and passes it explicitly in tests for determinism. Resolving here since the fix belongs in the upstream bead — this PR can land before #149 and inherit the tick on merge order.

File: web-mobile/js/admin_schedule.jsx [hunk 8]
```diff
@@ -849,6 +862,168 @@ function PerCourtBreakdown({ perCourtMinutes }) {
...
+  useEffectA(() => {
+    const timer = setInterval(() => setTick(t => t + 1), 60000);
+    return () => clearInterval(timer);
+  }, []);
+
+  const stats = useMemoA(
+    () => computeCourtPaceStats(byCourt, safeMatchDuration),
+    [byCourt, safeMatchDuration, tick]
+  );
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_schedule.jsx]: Fixed in aae8da64: dropped `useMemoA` and compute `stats = computeCourtPaceStats(byCourt, safeMatchDuration)` directly. Since `paceByCourt` is rebuilt on every parent render the memo dependency always changed anyway — no caching benefit was being gained. The 60-second `setTick` interval still forces re-renders so `computeCourtPaceStats` re-reads the current wall-clock time on each tick.

File: web-mobile/js/admin_schedule.jsx [hunk 9]
```diff
@@ -849,6 +862,168 @@ function PerCourtBreakdown({ perCourtMinutes }) {
...
+
+    return { court, completedCount, remainingCount, wallClockElapsedMin, estimatedRemainingMin, plannedRemainingMin, delta };
+  });
+}
+
+// CourtPacePanel — admin-only collapsible card showing per-court pace status
+// and a rebalancing suggestion. Never rendered in viewer or display views.
+function CourtPacePanel({ byCourt, safeMatchDuration }) {
+  const [open, setOpen] = useStateA(false);
+  const [tick, setTick] = useStateA(0);
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_schedule.jsx]: Fixed in aae8da64: changed to `const [, setTick] = useStateA(0)`. With `useMemoA` dropped the tick value was truly unread, so we now destructure only the setter. The re-render-based refresh contract is unchanged — `setTick(t => t + 1)` every 60 s triggers a re-render which re-invokes `computeCourtPaceStats` and picks up the current wall-clock time.

File: web-mobile/js/admin_schedule.jsx [hunk 10]
```diff
@@ -849,6 +862,168 @@ function PerCourtBreakdown({ perCourtMinutes }) {
...
+export function computeCourtPaceStats(byCourt, perMatchMinutes, nowMinutes) {
+  const ppm = perMatchMinutes > 0 ? perMatchMinutes : 3;
+  const nowMin = nowMinutes !== undefined ? nowMinutes : (() => {
+    const d = new Date();
+    return d.getHours() * 60 + d.getMinutes();
+  })();
+
+  return Object.entries(byCourt).map(([court, matches]) => {
+    const completedCount = matches.filter(m => m.status === "completed").length;
+    const remainingCount = matches.length - completedCount;
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_schedule.jsx]: Already handled in the current code: `computeCourtPaceStats` uses `m.status !== "scheduled" && m.status !== "running"` to count completed slots. Any other status (including kiken, fusenpai, forfeit) is treated as a consumed slot — same heuristic as the `courtOrder` sort that maps unknown statuses to the completed bucket.

File: web-mobile/js/admin_schedule.jsx [hunk 11]
```diff
@@ -214,6 +214,17 @@ function AdminSchedulePage({ tournament, onBack, onMoveCourt, onLogout, onViewer
...
 
+  // Pace stats must reflect the full per-court workload — the admin uses
+  // the panel to rebalance scheduling, not to inspect filter results.
+  // Build a separate by-court bucket from allMatches (still narrowed by
+  // the explicit ?court= scope so the panel can be reviewed one court at
+  // a time, but NOT by the ad-hoc player/dojo/competition filters).
+  const paceByCourt = {};
+  courts.forEach((cc) => paceByCourt[cc] = []);
+  filterMatchesByCourt(allMatches, effectiveCourt).forEach((m) => {
+    if (m.court && paceByCourt[m.court]) paceByCourt[m.court].push(m);
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_schedule.jsx]: Already fixed: `paceByCourt` starts from `courts.forEach((cc) => paceByCourt[cc] = [])` then extends for any `m.court` not in the configured list via `if (!paceByCourt[court]) paceByCourt[court] = []`. Stale or moved court assignments are therefore preserved in the pace panel rather than silently dropped.

File: web-mobile/js/admin_schedule.jsx [hunk 12]
```diff
@@ -214,6 +214,22 @@ function AdminSchedulePage({ tournament, onBack, onMoveCourt, onLogout, onViewer
...
+  // the explicit ?court= scope so the panel can be reviewed one court at
+  // a time, but NOT by the ad-hoc player/dojo/competition filters).
+  const paceByCourt = {};
+  courts.forEach((cc) => paceByCourt[cc] = []);
+  // Also bucket matches on courts not in the current config (stale/moved)
+  // so the pace panel stays accurate even after court list changes.
+  filterMatchesByCourt(allMatches, effectiveCourt).forEach((m) => {
+    if (m.court) {
+      if (!paceByCourt[m.court]) paceByCourt[m.court] = [];
+      paceByCourt[m.court].push(m);
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_schedule.jsx]: Already fixed: the `paceByCourt` builder now does `const court = (m.court || "").trim()` and skips the bucket when `court` is empty (`if (court) { ... }`). Whitespace-only labels are therefore treated as unassigned and cannot create phantom pace tiles.

File: web-mobile/js/admin_schedule.jsx [hunk 13]
```diff
@@ -957,10 +980,7 @@ export function CourtPacePanel({ byCourt, safeMatchDuration }) {
     return () => clearInterval(timer);
   }, [hasData]);
 
-  const stats = useMemoA(
-    () => computeCourtPaceStats(byCourt, safeMatchDuration),
-    [byCourt, safeMatchDuration, tick]
-  );
+  const stats = computeCourtPaceStats(byCourt, safeMatchDuration);
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_schedule.jsx:985]: Fixed in 88d44c6 — updated the doc comment above `computeCourtPaceStats`:
- Replaced "pure helper" with "deterministic when `nowMinutes` is supplied; non-deterministic (reads wall-clock via `new Date()`) when omitted"
- Replaced "forces the memo to re-invoke this helper" (stale after `useMemoA` removal) with "forces a re-render so this helper re-reads fresh wall-clock time on each tick"
